bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientAPIHandler.java

```diff
@@ -0,0 +1,401 @@
+        List<String> results = new ArrayList<>(things.size());
+        for (Thing t : things) {
+            AirGradientLocationHandler handler = (AirGradientLocationHandler) t.getHandler();
+            if (handler != null) {
+                results.add(handler.getLocationId());
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     * Return list of measures from AirGradient API.
+     *
+     * @return list of measures
+     */
+    public List<Measure> getMeasures() {
+        try {
+            ContentResponse response = httpClient.GET(generateMeasuresUrl());
+            String contentType = response.getMediaType();
+            logger.debug("Got measurements with status {}: {} ({})", response.getStatus(),
+                    response.getContentAsString(), contentType);
+            if (response.getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+                String stringResponse = response.getContentAsString().trim();
+
+                if (CONTENTTYPE_JSON.equals(contentType)) {
+                    return parseJson(stringResponse);
+                } else if (CONTENTTYPE_TEXT.equals(contentType)) {
+                    return parsePrometheus(stringResponse);
+                } else if (CONTENTTYPE_OPENMETRICS.equals(contentType)) {
+                    return parsePrometheus(stringResponse);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, response.getContentAsString());
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        return Collections.emptyList();
+    }
+
+    public void setLedMode(String serialNo, String mode) {
+        Request request = httpClient.newRequest(generateGetLedsModeUrl(serialNo));
+        request.method(HttpMethod.PUT);
+        request.header(HttpHeader.CONTENT_TYPE, CONTENTTYPE_JSON);
+        LedMode ledMode = new LedMode();
+        ledMode.mode = mode;
+        String modeJson = gson.toJson(ledMode);
+        logger.debug("Setting LEDS mode for {}: {}", serialNo, modeJson);
+        request.content(new StringContentProvider(CONTENTTYPE_JSON, modeJson, StandardCharsets.UTF_8));
+        try {
+            ContentResponse response = request.send();
+            logger.debug("Response from setting LEDs mode: {}", response.getStatus());
+            if (response.getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, response.getContentAsString());
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
```

Comment:
This is duplicate code to some other methods (like `calibrateCo2`) in this class. Maybe good to extract this and seperate the request creation (this method) and handling the request/response (new method)

Reply:
Thanks! That cleaned up the handler a lot!